### PR TITLE
Added namespaces to all files

### PR DESCRIPTION
**Pull request by sopho-s**
Haven't managed to get my compiler to work with this (really unsure why) but intellisense tells me it's fine (I wouldn't trust it but getting my compiler to work with this is a work in process 😢) if you want to compile it an check that would be great but I believe I haven't changed any code (hopefully)

### CODE REVIEW

**Comment by jochemste** [comment]
Firstly, thanks for the effort you are putting in! I have resolved the merge conflicts here on Github (apparently that is also possible). The code still has some issues, which you can see from the Github Action that failed (this builds the code in a Docker container with a Ubuntu image). This most likely is related to changed calls to the code that is now in namespaces. Try to have a look and see if you can see what is going wrong. If you update this branch, make sure to first pull the new changes into your local branch. I will review the changes and leave comments where I might have some feedback.

**Comment by sopho-s** [comment]
So the reason for this compilation error was due to me accidentally leaving a variable outside the namespace declaration

I would also like to ask, I'm trying to build with cmake (never used cmake before in my life) and I tried to run it and got this:

```Creating library C:/Users/nick/Documents/test/sneakyengine/_deps/sdl-build/Debug/SDL3.lib and ob
  ject C:/Users/nick/Documents/test/sneakyengine/_deps/sdl-build/Debug/SDL3.exp
  SDL3-shared.vcxproj -> C:\Users\nick\Documents\test\sneakyengine\bin\Debug\SDL3.dll
  Building Custom Rule C:/Users/nick/Documents/test/sneakyengine/src/logging/CMakeLists.txt
cl : command line error D8021: invalid numeric argument '/Wextra' [C:\Users\nick\Documents\test\sneak
yengine\src\logging\logger.vcxproj]
  Building Custom Rule C:/Users/nick/Documents/test/sneakyengine/_deps/googletest-src/googlemock/CMak
  eLists.txt
cl : command line error D8021: invalid numeric argument '/Wextra' [C:\Users\nick\Documents\test\sneak
yengine\_deps\googletest-build\googlemock\gmock.vcxproj]
  Building Custom Rule C:/Users/nick/Documents/test/sneakyengine/_deps/googletest-src/googlemock/CMak
  eLists.txt
cl : command line error D8021: invalid numeric argument '/Wextra' [C:\Users\nick\Documents\test\sneak
yengine\_deps\googletest-build\googlemock\gmock_main.vcxproj]
  Building Custom Rule C:/Users/nick/Documents/test/sneakyengine/_deps/googletest-src/googletest/CMak
  eLists.txt
cl : command line error D8021: invalid numeric argument '/Wextra' [C:\Users\nick\Documents\test\sneak
yengine\_deps\googletest-build\googletest\gtest.vcxproj]```

I searched it up and got [this](https://stackoverflow.com/questions/2274006/cmake-invalid-numeric-argument-wextra) but I don't want to edit any of the cmake files as I don't want to mess anything up

**Comment by jochemste** [comment]
That is related to compiler options working for Linux, but not for Windows. I have not tried building this for Windows yet, only for Linux so far, but I would definitely want to add this, so this info is helpful. I will create a new issue for that and fix it soon, so you could hopefully build it locally as well. By the way, CMake is used very often, both in industry and open source projects, so a good skill to have.

As for the compilation errors, I see that there are more cases of references to variables in namespaces, like `LogLevel`.

File: src/display/process_impl/display_proc_impl.cpp [hunk 1]
```diff
@@ -6,117 +6,120 @@
 
 #include <memory>
 #include <mutex>
+namespace display {
+  namespace display_process {
```
Comment (jochemste) [inline src/display/process_impl/display_proc_impl.cpp:10]: Thinking about how to call classes in this namespace (for example `display::display_process::DisplayProcessImpl()`) makes me feel like we could change this to have less repetition. So I would suggest to change `display_process` to `process`, so we can call `display::process::DisplayProcessImpl()`

File: src/display/process_impl/display_proc_impl.hpp [hunk 2]
```diff
@@ -1,11 +1,14 @@
 #include "process.hpp"
 
 #include <memory>
+namespace display {
+  namespace display_process {
```
Comment (jochemste) [inline src/display/process_impl/display_proc_impl.hpp:5]: Same here, I would suggest to change this to `process`

File: src/input/event/eventhandler_impl.cpp [hunk 3]
```diff
@@ -1,13 +1,17 @@
 #include "eventhandler_impl.hpp"
 
-EventHandlerImpl::EventHandlerImpl()
-    : m_inputpoller(input::InputPollerFactory().create()), m_stop(false) {}
+namespace input {
+  namespace event {
```
Comment (jochemste) [inline src/input/event/eventhandler_impl.cpp:4]: Same as with the other implementations, I do not think anything with `impl` in the filename/directoryname will need to be in the namespaces that are used externally (externally meaning outside of the module/submodule). Implementations are only referenced internally

File: src/input/event/eventhandler_impl.hpp [hunk 4]
```diff
@@ -4,16 +4,19 @@
 #include "event.hpp"
 
 #include "input.hpp"
+namespace input {
+  namespace event {
```
Comment (jochemste) [inline src/input/event/eventhandler_impl.hpp:8]: Same for this implementation

File: src/input/process_input/event_proc.cpp [hunk 5]
```diff
@@ -7,93 +7,97 @@
...
-  Log(LogLevel::debug) << LOG_END;
-}
-
-void EventHandlerProcessImpl::update_state(const ProcessState &state) {
-  Log(LogLevel::debug) << LOG_START;
-  const std::lock_guard<std::mutex> lock(m_state_mutex);
-  m_state = state;
-  Log(LogLevel::debug) << LOG_END;
+namespace input {
+  namespace process_input {
```
Comment (jochemste) [inline src/input/process_input/event_proc.cpp:11]: same as with `display_process`, I would shorten this to `process`

File: src/input/process_input/event_proc.hpp [hunk 6]
```diff
@@ -2,10 +2,14 @@
 
 #include <memory>
 
-class EventHandlerProcessFactory {
-public:
-  EventHandlerProcessFactory(){};
+namespace input {
+  namespace process_input {
```
Comment (jochemste) [inline src/input/process_input/event_proc.hpp:6]: process_input to process

File: src/logging/logging.hpp [hunk 7]
```diff
@@ -7,49 +7,51 @@
...
-  LogLevel m_loglevel_buffer;
-
-  /// The buffer to log
-  std::ostringstream m_buffer;
-
-  /// The implemented instance of the logger
-  std::unique_ptr<ILogInstance> m_logger;
-};
+namespace logging {
+  #define LOG_HEADER                \
```
Comment (jochemste) [inline src/logging/logging.hpp:11]: I would prefer to keep these definitions outside of the namespace, they are used often and using the namespace with these makes things hard to read quickly.

File: src/process/process/libs/scheduler/scheduler.cpp [hunk 8]
```diff
@@ -1,3 +1,7 @@
 #include "scheduler.hpp"
 
-namespace scheduler {} // namespace scheduler
+namespace process {
```
Comment (jochemste) [inline src/process/process/libs/scheduler/scheduler.cpp:3]: scheduler is a library to process, and should not be in the same namespace, so this namespace can be removed here.

File: src/process/process/libs/scheduler/scheduler.hpp [hunk 9]
```diff
@@ -7,75 +7,76 @@
 
 #include "logging.hpp"
 
-namespace scheduler {
+namespace process {
```
Comment (jochemste) [inline src/process/process/libs/scheduler/scheduler.hpp:10]: as mentioned in scheduler.cpp, I would prefer scheduler to not be in the process namespace

File: src/process/process/libs/threadpool/impl/default/threadpool_impl.cpp [hunk 10]
```diff
@@ -2,136 +2,138 @@
 
 #include "logging.hpp"
 
-namespace threadpool {
-Threadpool_impl::Threadpool_impl()
-    : Threadpool_impl(std::thread::hardware_concurrency()) {
-  Log(LogLevel::debug) << LOG_START;
-  Log(LogLevel::debug) << LOG_END;
-}
+namespace process {
```
Comment (jochemste) [inline src/process/process/libs/threadpool/impl/default/threadpool_impl.cpp:5]: Same as for scheduler, threadpool is a library and should not be in the process namespace

**Review by jochemste — CHANGES_REQUESTED** [review]
Overall very nice work. I still have some comments about what can be changed, so have a look when you get the chance. In general I would prefer to have `impl` files/classes/directories out of the namespaces that are used externally. Otherwise this could be confusing, since anything with `impl` is only used inside the module, not outside. Let me know if I can help. Windows support is on the way, so that should make it easier

**Comment by jochemste** [comment]
Windows build is now available. Have a look if you are able to try and compile

**Comment by sopho-s** [comment]
@jochemste Apologies if I cannot do much for a few days, I am a bit busy currently, but I will back to it straight on Sunday 😄

**Comment by sopho-s** [comment]
> Windows build is now available. Have a look if you are able to try and compile

I managed to get it working with your repository, I'll get the changes and pull it into my repo and try and fix the errors
